qiita_ware/processing_pipeline.py

```diff
@@ -254,6 +254,40 @@ def _get_preprocess_fasta_cmd(raw_data, prep_template, params):
                         "-o %s" % output_dir,
                         params_str])
     else:
+        len_seqs = len(seqs)
+        len_mapping_fps = len(mapping_fps)
+
+        if len_mapping_fps > len_seqs:
+            mapping_fps = [basename(m) for m in mapping_fps]
+            sffs = [basename(s) for s in sffs]
+            raise ValueError(
+                'Your prep template defines: "%s" but you only have "%s" as '
+                'files' % (', '.join(mapping_fps), ', '.join(sffs)))
+
+        if len_seqs != len_mapping_fps:
```

Comment:
if `len_seqs` is `==` to `len_mapping_fps` then `counter` will not be set triggering an exception on line 280